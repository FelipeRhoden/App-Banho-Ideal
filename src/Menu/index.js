import React from 'react';
import { View, Text, TextInput } from 'react-native';

import styles from './styles';

export default class Menu extends React.Component {
    constructor(props){
        props = props || {};
        props.m3 = props.m3 || 0;
        props.m3OnChange = props.m3OnChange || (() => {});
        props.KHw = props.KHw || 0;
        props.KHwOnChange = props.KHwOnChange || (() => {});
        props.Watts = props.Watts || 0;
        props.WattsOnChange = props.WattsOnChange || (() => {});
        super(props);
    }

    render(){
        return (
            <View >
                <Text>√Ågua</Text>
                <TextInput
                    onChangeText={text => this.props.m3OnChange(text)}
                    value={this.props.m3}
                />
                <Text>Luz</Text>
                <TextInput
                    onChangeText={text => this.props.KHwOnChange(text)}
                    value={this.props.KHw}
                />
                <Text>Chuveiro</Text>
                <TextInput
                    onChangeText={text => this.props.WattsOnChange(text)}
                    value={this.props.Watts}
                />
            </View>
        );
    }
}