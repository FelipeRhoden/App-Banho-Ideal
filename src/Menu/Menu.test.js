import React from "react";
import renderer from "react-test-renderer";

import Menu from './';

describe('<Menu />', function(){
    it('Deveria conter 3 props (m3, KHw, Watts)', ()=>{
        const menu = new Menu();

        expect(menu.props).toEqual(expect.objectContaining({
            m3: expect.anything(),
            m3OnChange: expect.anything(),
            KHw: expect.anything(),
            KHwOnChange: expect.anything(),
            Watts: expect.anything(),
            WattsOnChange: expect.anything(),
        }));
    });

    it('Deveria renderizar corretamente', ()=>{
        const menu = renderer.create(<Menu />).toJSON();

        expect(menu).toMatchSnapshot();
    })
});