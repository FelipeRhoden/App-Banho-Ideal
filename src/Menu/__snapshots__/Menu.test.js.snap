// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Deveria renderizar corretamente 1`] = `
<View>
  <Text>
    √Ågua
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
  <Text>
    Luz
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
  <Text>
    Chuveiro
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
  />
</View>
`;
